src/Dao/EnwikiDao.php

```diff
@@ -75,6 +97,37 @@ public function getUserDetails( $diff ) {
 
 
 	/**
+	 * Determine which of the given pages are dead
+	 *
+	 * @param $titles array Page titles
+	 * @return array the pages that are dead
+	 */
+	public function getDeadPages( $titles ) {
+		if ( !$titles ) {
+			return false;
+		}
+		$titles = array_map( function( $title ) {
```

Comment:
You don't need the closure here, just use `$titles = array_map( 'urlencode', $title );`